src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -133,11 +136,27 @@ private HttpResponse doApiCall(String url) {
 		}
 		return httpResponse;
 	}
-	private void getThePublishedCourseList(String url, CoursesForSubaccounts cfs) {
+	private void getThePublishedCourseList(String url, CoursesForSubaccounts cfs,HttpServletResponse response) {
 		M_log.debug("getThePublishedCourseList: Called");
 		HttpResponse httpResponse = doApiCall(url);
-		HttpEntity entity = httpResponse.getEntity();
 		ObjectMapper mapper = new ObjectMapper();
+		int statusCode = httpResponse.getStatusLine().getStatusCode();
+		if(statusCode!=200) {
```

Comment:
The code to generate an error is virtual the same in more than one method.  It needs to be factored out into a separate method.  If the exception message needs to change the message can be passed in as an argument.